q2_diversity/plugin_setup.py

```diff
@@ -103,6 +104,64 @@
                  " for all pairs of samples in a feature table.")
 )
 
+
+plugin.pipelines.register_function(
+    function=q2_diversity.beta_phylogenetic_meta,
+    inputs={'table': List[FeatureTable[Frequency |
+                          RelativeFrequency |
+                          PresenceAbsence]],
+            'phylogeny': List[Phylogeny[Rooted]]},
+    parameters={'metric': Str % Choices(beta.METRICS['PHYLO']['IMPL'] |
+                                        beta.METRICS['PHYLO']['UNIMPL']),
```

Comment:
The current implementation of `beta_phylogenetic_meta` in this PR doesn't actually have any discretely implement individual metrics ([for example](https://github.com/qiime2/q2-diversity-lib/blob/dead1b4ad531d19fd06516b844c20a59d9803495/q2_diversity_lib/beta.py#L182-L201)), if that is the intent, please remove the `IMPL` metrics:

```suggestion
    parameters={'metric': Str % Choices(beta.METRICS['PHYLO']['UNIMPL']),
```

Reply:
Not fully sure I follow. Unweighted, weighted, variance adjusted and generalized should be suitable for meta unifrac, and all of these are implemented in `unifrac`? I may be misunderstanding something here

Reply:
The reason we set up q2-diversity-lib is so that we can implement per-metric-qiime2-methods (like what I linked to above). This is great for provenance etc. The `IMPL` metrics are the list of metrics that have their own q2-diversity-lib "methods". The `UNIMPL` are metrics that don't have their own individual div-lib methods. If you peek at `beta` you'll see this illustrated more clearly:

https://github.com/qiime2/q2-diversity/blob/60de8d541e724dae7c453736599de78e6c1891fe/q2_diversity/_beta/_pipeline.py#L45-L52